mods/cnc/maps/nod07c/nod07c.lua

```diff
@@ -0,0 +1,202 @@
+
+targetsKilled = 0
+
+AutoGuard = function(guards)
+	Utils.Do(guards, function(guard)
+		Trigger.OnDamaged(guard, function(guard, attacker)
+			if not guard.IsDead and not attacker.IsDead then
+				guard.AttackMove(attacker.Location)
+			end
+		end)
+	end)
+end
+
+InsertNodUnits = function()
+	Media.PlaySpeechNotification(player, "Reinforce")
+	Reinforcements.Reinforce(player, { 'ltnk'}, { ReinforcementsTopSpawn.Location, ReinforcementsTankRally.Location }, 1)
+	local Engineers = Reinforcements.Reinforce(player, NodUnitsEngineers, { ReinforcementsTopSpawn.Location, ReinforcementsEngineersRally.Location }, 10)
+	Reinforcements.Reinforce(player, NodUnitsRockets, { ReinforcementsBottomSpawn.Location, ReinforcementsRocketsRally.Location }, 10)
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		Reinforcements.Reinforce(player, NodUnitsGunners, { ReinforcementsBottomSpawn.Location, ReinforcementsGunnersRally.Location }, 10)
+		Reinforcements.Reinforce(player, NodUnitsFlamers, { ReinforcementsTopSpawn.Location, ReinforcementsFlamersRally.Location }, 10)
+	end)
+	Trigger.AfterDelay(DateTime.Seconds(5), function()
+		local unitsA = Reinforcements.ReinforceWithTransport(player, 'tran.in', NodUnitsVehicles1, { GunboatRight.Location, ReinforcementsHelicopter1Rally.Location }, { GunboatRight.Location })[2]
+	
+		Trigger.AfterDelay(DateTime.Seconds(3), function()
+			local unitsB = Reinforcements.ReinforceWithTransport(player, 'tran.in', NodUnitsVehicles2, { GunboatRight.Location, ReinforcementsHelicopter2Rally.Location }, { GunboatRight.Location })[2]
+
+			Utils.Do(unitsB, function(unit)
+				unitsA[#unitsA + 1] = unit
+			end)
+
+			Trigger.OnAllKilled(unitsA, function()
+				if not defendersActive then
+					defendersActive = true
+					player.MarkFailedObjective(NodObjective3)
+				end
+			end)
+		end)
+	end)
+
+	Trigger.OnAllRemovedFromWorld(Engineers, function()
+		if not player.IsObjectiveCompleted(NodObjective1) then 
+			player.MarkFailedObjective(NodObjective1)
+		end
+	end)
+end
+
+DiscoveredSideEntrance = function(_,discoverer)
+	if not defendersActive then
+		defendersActive = true
+		player.MarkFailedObjective(NodObjective3)
+	end
+end
+
+DiscoveredMainEntrance = function(_,discoverer)
+	if not defendersActive then
+		SendDefenders(GDI2)
+	end
+end
+
+SendDefenders = function(team)
+	defendersActive = true
+	player.MarkCompletedObjective(NodObjective3)
+	Trigger.AfterDelay(DateTime.Seconds(3), function()
+		for type, amount in pairs(team.units) do
+			local count = 0
+			local actors = enemy.GetActorsByType(type)
+			Utils.Do(actors, function(actor)
+				if actor.IsIdle and count < amount then
+					actor.AttackMove(waypoint0.Location)
+					count = count + 1
+				end
+			end)
+		end
+	end)
+end
+
+SendGuards = function(team)
+	for type, amount in pairs(team.units) do
+		local count = 0
+		local actors = enemy.GetActorsByType(type)
+		Utils.Do(actors, function(actor)
+			if actor.IsIdle and count < amount then
```

Comment:
Instead of using those local `count` variables here and above you could just use `Utils.Take`.
For example
```
SendGuards = function(team)
	for type, amount in pairs(team.units) do
		local actors = Utils.Take(amount, enemy.GetActorsByType(type))
		Utils.Do(actors, function(actor)
			if actor.IsIdle then
				actor.Patrol(team.waypoints, true, DateTime.Seconds(25))
			end
		end)
	end
end
```
`Utils.Take` will even work when there are less elements in the table than you want to take.